libs/parallelism/lcos_local/include/hpx/lcos_local/channel.hpp

```diff
@@ -252,7 +252,14 @@ namespace hpx { namespace lcos { namespace local {
                 empty_ = false;
                 push_active_ = false;
             }
-            void set_deferred(T&& val) noexcept
+            void set_deferred(T&& val)
+            // CUDA versions less than 11.1 don't compile push_pt correctly if
+            // this is noexcept. hpx::util::result_of (and std::result_of)
+            // inside deferred_call does not detect that the call is valid, and
+            // compilation fails.
+#if !defined(HPX_CUDA_VERSION) || (HPX_CUDA_VERSION >= 1101)
```

Comment:
Could this be done using a feature test (see above)? We could define a `HPX_NOEXCEPT(...)` macro that hides all of this ugliness.

Reply:
See my comment about the other noexcept workaround, but I much prefer the explicit workaround here. There's no need to make `noexcept` conditional *everywhere* (even if it's just with certain nvcc versions) just because it doesn't work in this particular case. And if it wouldn't be `HPX_NOEXCEPT` everywhere `noexcept` is currently used I would at least find it confusing if only this member function would use `HPX_NOEXCEPT` and other places would use `noexcept`. This is just my preference though, and I'd be happy to consider the alternative if there's a good case for it.